### PR TITLE
Proxy to HistoricalRPCService

### PR DESCRIPTION
**Pull request by alecps**
- Minor edits to proxy out to the HistoricalRPCService based on cel2 time rather than bedrock block. 
- Adds support for TraceCall proxying to HistoricalRPCService

Fixes https://github.com/celo-org/celo-blockchain-planning/issues/386

Testing:
- Manually tested all rpc endpoints (see doc here https://www.notion.so/clabsco/HistoricalRPCService-Testing-Scratchpad-10aea6297b89809789bde885503a6210?showMoveTo=true&saveParent=true) 
- Working on adding unit tests, but have gotten stuck. See https://github.com/celo-org/op-geth/pull/238

### CODE REVIEW

File: eth/tracers/api.go
Comment (carterqw2) [inline eth/tracers/api.go:964]: Perhaps we can combine both conditionals.

**Review by karlb — APPROVED** [review]

File: eth/tracers/api.go
Comment (karlb) [reply eth/tracers/api.go:964]: I don't see a simple way to do it since we have three different results:
* Return result from local node
* Return proxies result
* Return `rpc.ErrNoHistoricalFallback`

Comment (piersy) [reply eth/tracers/api.go:964]: If you combined them then you'd have to repeat one of the conditions though, right?

File: params/config.go
Comment (piersy) [inline params/config.go]: This doesn't look right to me, IsPreCel2 will return true in situations where the cel2 fork was not set

Comment (karlb) [reply params/config.go]: Yes, Optimism has the `IsOptimism` check in addition. For us, it should probably be something like `c.Cel2Time != nil && !c.IsCel2(time)`

**Review by karlb — CHANGES_REQUESTED** [review]
Do we already understand why we get the "implement me" test failure?

**Comment by piersy** [comment]
> Do we already understand why we get the "implement me" test failure?

I don't, and I'm looking into that now

File: params/config.go
Comment (piersy) [reply params/config.go]: I'm just testing this now

**Comment by karlb** [comment]
Fixing the IsPreCel2 condition fixes the two failing tests locally. Let's see if CI agrees.

**Comment by piersy** [comment]
> Fixing the IsPreCel2 condition fixes the two failing tests locally. Let's see if CI agrees.

No all the pre-existing historical rpc tests are failing because they don't have cel2 set in their test config. But setting cel2 is causing further problems (at least in `TestEthClientHistoricalBackend`)

File: params/config.go
Comment (gastonponti) [inline params/config.go:773]: We have a config.Celo now due to the baseFeeFloor, if we want to mimic the same logic that the optimism's team is doing.

**Comment by karlb** [comment]
I forced-pushed to rebase on https://github.com/celo-org/op-geth/pull/239, which helped getting the tests to pass. The only actually new commit in the recent push is https://github.com/celo-org/op-geth/pull/236/commits/cd9a87fc64fda450a20b7e039669943d84ce2277. With that the tests should pass and this PR should be ready.

**Review by karlb — APPROVED** [review]
I'll approve to undo my "request changes", but someone else should review before merging, because I did the last changes myself.

**Pull request merged**
